element/index.js

```diff
@@ -32,9 +32,34 @@ export { render };
 export { Component };
 
 /**
+ * Creates a copy of an element with extended props.
+ *
+ * @param  {WPElement} element Element
+ * @param  {?Object}   props   Props to apply to cloned element
+ * @return {WPElement}         Cloned element
+ */
+export { cloneElement };
```

Comment:
To clarify my point, I don't see much value in abstracting React because we're using a lot of React features: Element, Component (refs, state, lifecycle), cloneElement, renderToString, render. Which is close to be the whole React.

Also, and this is probably the main reason for me. If we want to use a library like `react-textarea-autosize` or `react-tooltip` or any library dependent on React, we can do so (we're already doing it) technically and it'll work because we're using React but what if we switch our `wp.element` implementation to a custom React alternative, nothing guarantees that the libraries we're using are only using the React features we whitelist on our implementation. And this IMO should force us to avoid using any React third-party library because "logically" they're not dependent on our abstraction but they're dependent on React (which is a larger library).